base/applications/cmdutils/taskkill/taskkill.c

```diff
@@ -348,64 +533,24 @@ static int terminate_processes(void)
 
             if (!TerminateProcess(process, 0))
             {
-                taskkill_message_printfW(STRING_TERMINATE_FAILED, task_list[i]);
+                taskkill_message_printfW(STRING_TERMINATE_FAILED, process_name);
                 status_code = 1;
                 CloseHandle(process);
                 continue;
             }
 
-            taskkill_message_printfW(STRING_TERM_PID_SEARCH, pid);
-            CloseHandle(process);
-        }
-        else
-        {
-            DWORD index;
-            BOOL found_process = FALSE;
-
-            for (index = 0; index < pid_list_size; index++)
+            if (is_numeric)
             {
-                WCHAR process_name[MAX_PATH];
-
-                if (get_process_name_from_pid(pid_list[index], process_name, MAX_PATH) &&
-                    !strcmpiW(process_name, task_list[i]))
-                {
-                    HANDLE process;
-
-                    if (pid_list[index] == self_pid)
-                    {
-                        taskkill_message(STRING_SELF_TERMINATION);
-                        status_code = 1;
-                        continue;
-                    }
-
-                    process = OpenProcess(PROCESS_TERMINATE, FALSE, pid_list[index]);
-                    if (!process)
-                    {
-                        taskkill_message_printfW(STRING_SEARCH_FAILED, task_list[i]);
-                        status_code = 128;
-                        continue;
-                    }
-
-                    if (!TerminateProcess(process, 0))
-                    {
-                        taskkill_message_printfW(STRING_TERMINATE_FAILED, task_list[i]);
-                        status_code = 1;
-                        CloseHandle(process);
-                        continue;
-                    }
-
-                    found_process = TRUE;
-                    taskkill_message_printfW(STRING_TERM_PROC_SEARCH, task_list[i], pid_list[index]);
-                    CloseHandle(process);
-                }
+                taskkill_message_printfW(STRING_TERM_PID_SEARCH, pid);
             }
-
-            if (!found_process)
+            else
             {
-                taskkill_message_printfW(STRING_SEARCH_FAILED, task_list[i]);
-                status_code = 128;
+                taskkill_message_printfW(STRING_TERM_PROC_SEARCH, process_name, pid);
             }
+            CloseHandle(process);
         }
+
+        HeapFree(GetProcessHeap(), 0, pkill_list);
```

Comment:
See my comment in the other function concerning `pkill_list` (it also applies there).